app/Models/LdapAdConfiguration.php

```diff
@@ -0,0 +1,245 @@
+                if ('ldap_pword' === $key) {
+                    try {
+                        return decrypt($item);
+                    } catch (Exception $e) {
+                        throw new Exception('Your app key has changed! Could not decrypt LDAP password using your current app key, so LDAP authentication has been disabled. Login with a local account, update the LDAP password and re-enable it in Admin > Settings.');
+                    }
+                }
+
+                return $item;
+            });
+
+        return $ldapSettings;
+    }
+
+    /**
+     * Set the server certificate environment variable.
+     *
+     * @author Wes Hulette <jwhulette@gmail.com>
+     *
+     * @since 5.0.0
+     */
+    private function certificateCheck(): void
+    {
+        // If we are ignoring the SSL cert we need to setup the environment variable
+        // before we create the connection
+        if ($this->ldapSettings['ldap_server_cert_ignore']) {
+            putenv('LDAPTLS_REQCERT=never');
+        }
+
+        // If the user specifies where CA Certs are, make sure to use them
+        if (env('LDAPTLS_CACERT')) {
+            putenv('LDAPTLS_CACERT='.env('LDAPTLS_CACERT'));
+        }
+    }
+
+    /**
+     * Set the Adlap2 connection configuration values based on SnipeIT settings.
+     *
+     * @author Wes Hulette <jwhulette@gmail.com>
+     *
+     * @since 5.0.0
+     * 
+     * @return array
+     */
+    private function setLdapConnectionConfiguration(): array
+    {
+        // Create the configuration array.
+        return [
+            // Mandatory Configuration Options
+            'hosts'            => $this->getServerUrlBase(),
+            'base_dn'          => $this->ldapSettings['ldap_basedn'],
+            'username'         => $this->ldapSettings['ldap_uname'],
+            'password'         => $this->ldapSettings['ldap_pword'],
+
+            // Optional Configuration Options
+            'schema'           => $this->getSchema(),
+            'account_prefix'   => '',
+            'account_suffix'   => '',
+            'port'             => $this->getPort(),
+            'follow_referrals' => false,
+            'use_ssl'          => $this->isSsl(),
+            'use_tls'          => $this->ldapSettings['ldap_tls'],
+            'version'          => $this->ldapSettings['ldap_version'] ?? self::DEFAULT_LDAP_VERSION,
+            'timeout'          => self::CONNECTION_TIMEOUT,
+
+            // Custom LDAP Options
+            'custom_options'   => [
+                // See: http://php.net/ldap_set_option
+                // LDAP_OPT_X_TLS_REQUIRE_CERT => LDAP_OPT_X_TLS_HARD,
+            ],
+        ];
+    }
+
+    /**
+     * Get the schema to use for the connection.
+     *
+     * @author Wes Hulette <jwhulette@gmail.com>
+     *
+     * @since 5.0.0
+     *
+     * @return string
+     */
+    private function getSchema(): string
+    {
+        $schema = \Adldap\Schemas\OpenLDAP::class;
```

Comment:
Since we force our users to specify exactly what their schema is, I don't really know how this might change things, but it's probably good going forward, if not used already.

Reply:
The ADLDAP2 framework requires a schema for configuration. One of the schema's it supports is FreeIPA, which would be good to add in later.